src/pip/_internal/resolution/resolvelib/factory.py

```diff
@@ -603,7 +603,12 @@ def _report_single_requirement_conflict(
 
         cands = self._finder.find_all_candidates(req.project_name)
         skipped_by_requires_python = self._finder.requires_python_skipped_reasons()
-        versions = [str(v) for v in sorted({c.version for c in cands})]
+        versions = [
+            str(v)
+            for v in sorted(
+                {c.version for c in cands if not getattr(c.link, "is_yanked", False)}
+            )
+        ]
 
```

Comment:
@uranusjr how does this look?

```suggestion
        yanked_versions = [
            str(v)
            for v in sorted(
                {c.version for c in cands if getattr(c.link, "is_yanked", False)}
            )
        ]

        if yanked_versions:
            logger.critical(
                "Ignored the following yanked versions: %s",
                ", ".join(yanked_versions) or "none"
            )

```

Reply:
implemented ^